src/uploader.coffee

```diff
@@ -7,11 +7,15 @@ class Uploader extends EventEmitter
   # Constructor
   constructor: ({accessKey, secretKey, sessionToken, region, stream, objectName, objectParams, bucket, partSize, maxBufferSize, waitForPartAttempts, waitTime, debug}, @cb) ->
     super()
-    aws.config.update
-      accessKeyId:     accessKey
-      secretAccessKey: secretKey
-      sessionToken:    sessionToken
-      region:          region if region
+
+    if accessKey?
```

Comment:
you don't need use `accessKey?` just `accessKey` but on EC2 instances are set in environment variables?